### DISCUSSION TITLE
Igir show and tell megathread!

### DISCUSSION THREAD

**Discussion by emmercm** [emmercm/igir#1000] [Show and tell]
I created the "[personal usage](https://igir.io/usage/personal/)" documentation page to show off how I personally use `igir` with my own ROM collection. But that page only shows one person's needs, when so many other use cases could be solved with `igir`.

**I would love to see how you use `igir` with your own ROM collection!**

Some information that would be helpful to include:

- The entire set of commands and options used, including if you have to run `igir` multiple times
- Your process for initially sorting your collection
- Your process for incrementally updating your collection
- What devices (flash cards, emulation handhelds, etc.) you use with your collection

_To keep discussions clear, please use separate comments for separate use cases, and make use of the "reply" feature to respond to someone's use case. Thank you!_

**Comment by tnvu** [comment]
I'm still figuring out my workflow, but for now.

1. I have the ROMs I want to organize.
2. I run them through igir move zip test clean report
3. I torrentzip the output from igir.
4. I take the output of igir and import them into ROMVault (I like the interface better and it's a little more proven). This step tends to be fast since the files are (usually) properly prepared.

I especially like that there is command line support, parallel processing, and more supported file formats (rar, etc). I also like that it's opensource and makes it easy for me to find where potential problems might be. Also the developer is super responsive!

**Reply by emmercm** [reply]
Thank you for the message! I'd love to dig in on a few points:

- Is there a reason you prefer TorrentZip? Is it because of the first-class support in RomVault? I see the value for people joining MAME torrents, but I'm less sure about general collection sorting.
- What features do you use in RomVault after Igir has sorted your ROMs?

**Reply by tnvu** [reply]
I use TorrentZip primarily because of the first-class support in RomVault. It just makes my collection consistent with a lot of what's out there. I can checksum the zip and get the result without having to unpack it and then compare the file. It saves a step.

As for RomVault, post Igir... it's primarily the UI. Being able to visually see what's going on is pretty convenient. I know Igir isn't designed with that type of UI functionality in mind, so it's just a little easier to use than scanning through fixDATs.

The secondary use is to verify the work done by Igir. It caught #1001 when I noticed it was recompressing the zips produced by Igir. Just a sanity check to run it against a second, more mature tool.

**Comment by IDmedia** [comment]
I love Igir, it's just a fantastic tool and I've made a proof-of-concept frontend for it.

The way it works is that I create a directory (e.g. "naomi2") with a subdirectory called "dats" and place all the relevant dat files in that directory for that system. Then I refresh the page and it lists all systems (directories) with dat files and automatically makes a report of missing roms based on your preferences. It also allows me to show the report as a HTML table, download missing roms from a site I won't mention here and start processing the current platform by pressing "process".

For now it's a "poor mans" *arr system for roms...
However, due to #1570 it's not working very well for some systems unfortunately.

Screenshots:
![image](https://github.com/user-attachments/assets/297c067e-573f-4958-919f-b236e8b1d163)
![image](https://github.com/user-attachments/assets/c3bed5e9-7093-44df-9025-ef63514e2652)

When/if the bug is fixed, I'll clean up the UI and make it pretty and easy to deploy using Docker. I think of this as an extension for RomM that is able to keep track of roms I have and which I'm still missing for a complete 1G1R collection. I make folders with the same naming convention as RomM supports so it parses and displays the roms without issues.


Once this is done I'm going to make a integration between RomM and RetroBat and I think that would be one of the best solutions for playing roms while storing them on a central server. More info: https://github.com/rommapp/romm/discussions/1665#discussioncomment-13469801

**Comment by TheRealBenForce** [comment]
Today is day 1 but this seems like an amazing tool. Hats off to all the contributors. I plan to make a docker image that runs each night to keep things tidy for me. I'm open to suggestions. Maybe I'm overengineering this but I think what I'd like is:

/roms-master (main collection zipped)
/roms-1g1r (trimmed down version using links. Not sure I need this but I thought it might make the next steps faster if I used it as my input for smaller collections.)
/roms-curated (links for 1g1r with filename filter excludes, hacks, translations, etc. I may have some 1g1r overrides but i'm not really sure how I would do that yet. )
/roms-{device|size} (links for specific device or size target)

Pretty simple container. 
```
version: '2'

services:
  igir:
    container_name: igir
    image: "mcr.microsoft.com/devcontainers/javascript-node:20-bullseye"
    restart: unless-stopped
    command: sleep infinity
    volumes:
      - '${MEDIA_SHARE}/emulation/dats/:/dats'
      - '${MEDIA_SHARE}/emulation/roms/:/roms'
```

Until I figure out how I like it to build my own docker image, I'm just installing igir manually then working on these scripts:
```
ROMS_REPORT="/roms/report %dddd, %MMMM %Do %YYYY, %h:%mm:%ss %a.csv"

CASINO="billiards|blackjack|Caesars Palace|casino|poker|Vegas"
CLASSICS="Arcade Classic|Arcade Hits|Atari|Konami Collector|Konami GB Collection|Namco Museum|Space Invaders *GB*"
COLLECTIONS="collection|Games in One|Game & Watch Gallery|Game Boy Gallery|Game Pack|In One|In-One|In-1|Ultimate * Games|Williams Arcade"
GAME_SHOWS="Familyl Feud|Jeopardy|Wheel of Fortune|Who Wants to Be a Millionaire"
INFERIOR_REMAKES="Super Mario Advance"
KIDS="Antz|Army Men|barbie|Bratz|Cabbage Patch kids|Digimon|Disney|Dogz|Dora the Explorer|Fairly Odd Parents|Fisher-Price|Great Waldo|Harry Potter|Kim Possible|Nick Toons|Petz|Scooby-Doo|Sesame Street|Shrek|Strawberry Shortcake|That's So Raven|Trollz|Winnie the Pooh|Yu-Gi-Oh"
OTHER="Pac-Man *"
SPORT="College Football|ESPN|F1 *|F-1 *|FIFA|MLB|NASCAR|NBA|NCAA|NFL|NHL|PGA|TNN"
SPORT_SERIES="Backyard Baseball|Bases Loaded|Big Game Hunter|Bill Elliot|Bill Walsh|Bulls v|Championship *|David Beckham|Davis Cup|Ferrari Grand|Formula One|Frank Thomas|George Foreman|HardBall|Hit the Ice|International *|Jack Nicklaus|Jimmy Connors|Jimmy White|Ken Griffey Jr|Madden|Michael Andretti|Muhammed Ali|Nigel Mansell|Olympic|Premier Manager|R.B.I.|RealSports|Roger clemens|Sports Illustrated|Summer Games|Tecmo|Tiger Woods|Tripple Play|Troy Aikman|WCW|WWE|Wimbledon|Winter Games|World Championship|World Class|World Cup|World Series|Zidane"
TV="Ben Ten|Kim Possible|Lizzy McGuire|Mary-Kate & Ashley|Mary-Kate and Ashley|SpongeBob"

PATTERN="${CASINO}|${CLASSICS}|${COLLECTIONS}|${GAME_SHOWS}|${INFERIOR_REMAKES}|${KIDS}|${OTHER}|${SPORT}|${SPORT_SERIES}|${TV}"


# Cartridge-based consoles, 1st-5th generations
npx --yes igir@latest move zip test clean report \
  --dat "/dats/*" \
  --dat-name-regex-exclude "/encrypted|source code/i" \
  --input "/roms/roms-to-sort" \
  --input-checksum-max CRC32 \
  --input-checksum-archives never \
  --output "/roms/roms-master" \
  --dir-dat-name \
  --overwrite-invalid \
  --zip-exclude "*.{chd,iso}" \
  --reader-threads 4 \
  --report-output "$ROMS_REPORT" \
  -v

npx --yes igir@latest link test clean report \
  --dat "/dats/*" \
  --dat-name-regex-exclude "/encrypted|source code/i" \
  --input "/roms/roms-master" \
  --input-checksum-max CRC32 \
  --input-checksum-archives never \
  --output "/roms/roms-1g1r" \
  --dir-dat-name \
  --overwrite-invalid \
  --reader-threads 4 \
  --single \
  --prefer-language EN \
  --prefer-region USA,WORLD,EUR,JPN \
  --prefer-revision newer \
  --report-output "$ROMS_REPORT" \
  -v

npx --yes igir@latest link test clean report \
  --dat "/dats/*" \
  --dat-name-regex-exclude "/encrypted|source code/i" \
  --input "/roms/roms-1g1r" \
  --input-checksum-max CRC32 \
  --input-checksum-archives never \
  --output "/roms/roms-curated" \
  --dir-dat-name \
  --overwrite-invalid \
  --reader-threads 4 \
  --single \
  --prefer-language EN \
  --prefer-region USA,WORLD,EUR,JPN \
  --prefer-revision newer \
  --filter-regex-exclude "$PATTERN" \
  --report-output "$ROMS_REPORT" \
  -v
```

**Comment by kencinder** [comment]
I moved to Linux from Windows and had been using clrmamepro for a long long time
I didn't want to deal with running it via Wine and had already been pointed in the direction of IGIR before switching OS'

I wanted a way to handle single systems at a time from one place, so I decided to start making my own menu setup in a script, which started out pretty basic and klunky for adding systems, and has now turned into a full dynamic setup where I can do single systems, entire platforms or all of them at once with 7z support too.

Now I just have to add a couple variables, copy and paste a config entry and adjust it, and a new system is added to the menu

https://github.com/kencinder/igirst
